README.md

```diff
@@ -728,6 +730,36 @@ func main() {
 }
 ```
 
+## Generics
```

Comment:
I would separate this into a few parts: type parameters, constraints, and generic types.